challenge.py

```diff
@@ -0,0 +1,117 @@
+import os
+import re
+import numpy as np
+
+List_calls = "calls.txt"
+List_files = "list_files.txt"
+result_stat = "stat.txt"
+flag_file = 1
+flag_calls = 1
+def result(filename):
+  
+  #local variable declaration
+  deleted = 0 
+  added = 0 
+  region = 0 
+  
+  #Regular expression declaration 
+  regex_FileName = '^diff --git a.+[ ]b/(.+?)\n'
+  regex_Del = "^-(?!--).*"
+  regex_Add = "^\+(?!\+\+).*"
+  regex_region = "^@@ -\d+(?:,\d+)? \+\d+(?:,\d+)?\ @@[ ]?.*"
+  regex_functionCall = "([\w]+)(?=\()"
+  #catch les nom des fonctions avec le regex, les mettres dans une liste, ensuite appliquer la fonction de l'occurence
+  regex_FileNamePattern = re.compile(regex_FileName)
+  regex_DelPattern = re.compile(regex_Del,re.DOTALL)
+  regex_AddPattern = re.compile(regex_Add,re.DOTALL)
+  regex_regionPattern = re.compile(regex_region)
+  regex_functionCallPattern = re.compile(regex_functionCall)
+  calls = []
+  Files_array = np.array([])
+  #parcour each line of diff file
+  for line in open(filename):
+    #to optimize we did test to be sure we use a useful line
+    if ( line.startswith("diff") ):
+      #check if we got a match with regex file name
+      for match in re.findall(regex_FileNamePattern, line):
+        Files_array = np.append(Files_array, match)
+    elif ( line.startswith("@@") ):
+      for match in re.findall(regex_regionPattern, line):
+        region = region + 1
+    elif ( line.startswith("+++") or line.startswith("---") or line.startswith("index") or line.startswith("deleted") or line.startswith("new")):
+      pass
+    elif ( line.startswith("-") ):
+      for match in re.findall(regex_functionCallPattern, line):
+        if match not in ('if', 'while', 'for', 'switch'):
+          calls.append(match)
+      for match in re.findall(regex_DelPattern, line):
+        deleted = deleted + 1
+    elif ( line.startswith("+") ):
+      for match in re.findall(regex_functionCallPattern, line):
+        if match not in ('if', 'while', 'for', 'switch'):
+          calls.append(match)
+      for match in re.findall(regex_AddPattern, line):
+        added = added + 1
+  return [Files_array,deleted,added,region,calls]
+
+
+if __name__ == "__main__":
+  try:
+    os.remove(List_files)
```

Comment:
Considering that any file may be missing, they should be checked separately.

OTOH, thay will be wiped when written over by `file.write`, removing them is unnecessary here